pkg/reconciler/v1alpha1/revision/resources/deploy_test.go

```diff
@@ -39,12 +39,219 @@ import (
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{FieldPath: "metadata.name"},
+			},
+		}, {
+			Name: "SERVING_LOGGING_CONFIG",
+			// No logging configuration
+		}, {
+			Name: "SERVING_LOGGING_LEVEL",
+			// No logging level
+		}, {
+			Name:  "USER_PORT",
+			Value: "8080",
+		}, {
+			Name:  "SYSTEM_NAMESPACE",
+			Value: system.Namespace(),
+		}},
+	}
+
+	defaultFluentdContainer = &corev1.Container{
+		Name:      FluentdContainerName,
+		Image:     "indiana:jones",
+		Resources: fluentdResources,
+		Env: []corev1.EnvVar{{
+			Name:  "FLUENTD_ARGS",
+			Value: "--no-supervisor -q",
+		}, {
+			Name:  "SERVING_CONTAINER_NAME",
+			Value: UserContainerName,
+		}, {
+			Name: "SERVING_CONFIGURATION",
+			// No owner reference
+		}, {
+			Name:  "SERVING_REVISION",
+			Value: "bar",
+		}, {
+			Name:  "SERVING_NAMESPACE",
+			Value: "foo",
+		}, {
+			Name: "SERVING_POD_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.name",
+				},
+			},
+		}},
+		VolumeMounts: fluentdVolumeMounts,
+	}
+
+	defaultPodSpec = &corev1.PodSpec{
+		Volumes:                       []corev1.Volume{varLogVolume},
+		TerminationGracePeriodSeconds: refInt64(45),
+	}
+
+	defaultDeployment = &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "foo",
+			Name:      "bar-deployment",
+			Labels: map[string]string{
+				serving.RevisionLabelKey: "bar",
+				serving.RevisionUID:      "1234",
+				AppLabelKey:              "bar",
+			},
+			Annotations: map[string]string{},
+			OwnerReferences: []metav1.OwnerReference{{
+				APIVersion:         v1alpha1.SchemeGroupVersion.String(),
+				Kind:               "Revision",
+				Name:               "bar",
+				UID:                "1234",
+				Controller:         &boolTrue,
+				BlockOwnerDeletion: &boolTrue,
+			}},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &one,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					serving.RevisionUID: "1234",
+				},
+			},
+			ProgressDeadlineSeconds: &ProgressDeadlineSeconds,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						serving.RevisionLabelKey: "bar",
+						serving.RevisionUID:      "1234",
+						AppLabelKey:              "bar",
+					},
+					Annotations: map[string]string{
+						sidecarIstioInjectAnnotation: "true",
+					},
+				},
+				// Spec: filled in by makePodSpec
+			},
+		},
+	}
 )
 
 func refInt64(num int64) *int64 {
 	return &num
 }
 
+type containerOpt func(*corev1.Container)
+type podSpecOpt func(*corev1.PodSpec)
+type deploymentOpt func(*appsv1.Deployment)
+
+func makeTestContainer(defaultContainer *corev1.Container, opts ...containerOpt) *corev1.Container {
+	container := defaultContainer.DeepCopy()
+	for _, opt := range opts {
+		opt(container)
+	}
+	return container
+}
+
+func makeTestUserContainer(opts ...containerOpt) *corev1.Container {
```

Comment:
tests with similar helpers usually drop the `make` and `Test` prefix.

In this case you could get away with:

```
func container(...)
func queueContainer(...)
func fluentdContainer(...)
func podSpec(...)
func deployment(...)
```

Which reads a bit more fuent, especially in conjunction with the other naming convention proposed:

```golang
queueContainer(withEnvVar("PORT", "8080"))
```

For example.

Reply:
Done.